numba/parfors/array_analysis.py

```diff
@@ -2997,9 +3006,21 @@ def _insert_runtime_broadcast_call(self, scope, loc, arrs, max_dim):
             ir.Assign(value=func_call, target=runtime_broadcast_shape, loc=loc)
         )
 
+        # Define the output in the equiv_set for the left hand side.
+        # These should be considered equivalent because we produced a
+        # runtime check that verifies this.
+        needs_define = equiv_set.insert_equiv(runtime_broadcast_shape, *arrs)
```

Comment:
I don't think this is correct, but I need a way to ensure the `runtime_broadcast_shape` is selected by the lhs of array_expr. This gets `test_2array_1_control_flow`, but again I don't think its correct.